### PR TITLE
feat[Api-docs] : open config & manage schema url for AsyncAPI

### PR DESCRIPTION
**Pull request by fb33**
This configuration manages AsyncApi-react component. (see https://github.com/asyncapi/asyncapi-react#props)

A resolver for AsyncApi parser is also added, so that a schema definition can be fetched from an http/s URL

This is linked to the issue : https://github.com/backstage/backstage/issues/7571

### CODE REVIEW

File: plugins/api-docs/src/components/AsyncApiDefinitionWidget/AsyncApiDefinitionWidget.tsx [hunk 1]
```diff
@@ -134,12 +133,46 @@ type Props = {
...
+  canRead(file: any) {
+    if ((this.canReadValue as any) instanceof Boolean) {
+      return this.canReadValue;
+    }
+    return file.url.match(new RegExp((this.canReadValue as unknown) as string));
+  },
+  fetchOptions: {},
+
+  async read(file: any) {
+    // @ts-ignore
```
Comment (Rugvip) [inline plugins/api-docs/src/components/AsyncApiDefinitionWidget/AsyncApiDefinitionWidget.tsx:148]: Avoid using @ts-ignore if possible, it's better to cast to `any`. What is it that is being ignored here?

File: plugins/api-docs/package.json [hunk 2]
```diff
@@ -37,9 +37,11 @@
     "@backstage/plugin-catalog": "^0.7.2",
     "@backstage/plugin-catalog-react": "^0.6.1",
     "@backstage/theme": "^0.2.11",
+    "@backstage/types": "^0.1.1",
     "@material-ui/core": "^4.12.2",
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "4.0.0-alpha.57",
+    "@spotify/prettier-config": "^12.0.0",
```
Comment (Rugvip) [inline plugins/api-docs/package.json:44]: This shouldn't be here

File: app-config.yaml [hunk 3]
```diff
@@ -450,3 +450,22 @@ azureDevOps:
   host: dev.azure.com
   token: my-token
   organization: my-company
+
+apiDocs:
```
[the remark this answers is not in the document]
Comment (Rugvip) [reply app-config.yaml:454]: +1 to the concern of this not being a great use-case for app-config, and that it perhaps would be better to handle this through overriding the `apiDocsConfigRef` implementation. Is that something you could have a look at @fb33?

**Review by Rugvip — COMMENTED** [review]
Thanks! 👍 

Looks good, but would like to see if we could avoid managing all of this through static config

File: app-config.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (fb33) [reply app-config.yaml:454]: Hello @Rugvip 
I managed to handle this through overriding the apiDocsConfigRef implementation.
```
    createApiFactory({
        api: apiDocsConfigRef,
        deps: {},
        factory: () => {
            // load the default widgets
            const definitionWidgets = defaultDefinitionWidgets();
            return {
                "getApiDefinitionWidget": (apiEntity: ApiEntity) => {
                    if (apiEntity.spec.type === 'asyncapi') {
                        return {
                            type: 'asyncapi',
                            title: 'AsyncAPI',
                            rawLanguage: 'yaml',
                            component: definition => <AsyncApiDefinitionWidget definition={definition} />
                        } as ApiDefinitionWidget
                    }

                    // fallback to the defaults
                    return definitionWidgets.find(d => d.type === apiEntity.spec.type);
                },
            };
        },
    }),
```
I override AsyncApiDefinitionWidget to define the config attribut of <AsyncApi >
There is another discussion here https://github.com/backstage/backstage/issues/7571#issuecomment-957152688

**Comment by dweber019** [comment]
If you like I can provide a PR with only the fetchResolver (no config or styles).
What do you think @Rugvip @Fox32 ?

**Pull request closed without merging**